packages/fxa-auth-server/test/local/routes/password.js

```diff
@@ -1010,6 +1010,70 @@ describe('/password', () => {
         );
       });
     });
+
+    it('upgrades to v2', async () => {
+      const uid = uuid.v4({}, Buffer.alloc(16)).toString('hex');
+      const mockDB = mocks.mockDB({
+        email: TEST_EMAIL,
+        uid,
+        // Signals that v1 password has not changed, and we have
+        // a password upgrade scenario
+        isPasswordMatchV1: true,
+      });
+      const mockPush = mocks.mockPush();
+      const mockMailer = {
+        sendPasswordChangedEmail: sinon.spy(() => {
+          return Promise.resolve();
+        }),
+      };
+      const mockLog = mocks.mockLog();
+      const mockRequest = mocks.mockRequest({
+        credentials: {
+          uid: uid,
+        },
+        payload: {
+          authPW: crypto.randomBytes(32).toString('hex'),
+          wrapKb: crypto.randomBytes(32).toString('hex'),
+          authPWVersion2: crypto.randomBytes(32).toString('hex'),
+          wrapKbVersion2: crypto.randomBytes(32).toString('hex'),
+          clientSalt:
+            'identity.mozilla.com/picl/v1/quickStretchV2:0123456789abcdef0123456789abcdef',
+          sessionToken: crypto.randomBytes(32).toString('hex'),
+        },
+        query: {
+          keys: 'true',
+        },
+        log: mockLog,
+      });
+
+      const passwordRoutes = makeRoutes({
+        config: {
+          domain: 'wibble',
+          smtp: {},
+          passwordForgotOtp: { enabled: false, digits: 8 },
+        },
+        db: mockDB,
+        push: mockPush,
+        mailer: mockMailer,
+        log: mockLog,
+      });
+
+      return runRoute(
+        passwordRoutes,
+        '/password/change/finish',
+        mockRequest
+      ).then((response) => {
+        assert.equal(mockDB.deletePasswordChangeToken.callCount, 1);
+        assert.equal(mockDB.resetAccount.callCount, 1);
```

Comment:
It'd be good to get a second set of eyes on this. 

I'm hoping we can call this and **not** notify RPs as long we know the v1 password is intact. However, this call does result in a delete in redis for the account,. and I'm not entirely sure this is okay. Perhaps the Redis delete needs to be skipped in an upgrade scenario?

Reply:
Oh interesting, the `resetAccount` function revokes [session tokens, devices, and more ](github.com/mozilla/fxa/blob/365d747e7223d02cc3f4976c86a37d9e47fb0ed5/packages/db-migrations/databases/fxa/patches/patch-144-145.sql#L184). It seems to me that we should have a custom `resetAccount` call that just updates the key stretch values.

Currently, the change password is a user initiated action (maybe from an email saying change password), so revoking all these tokens made sense and the user would want to be signed out everywhere. Since the key stretch upgrade is done siliently, maybe it makes sense to not sign them out.

Reply:
I agree, I'll update this so it doesn't destroy the session tokens, devices, etc...